sql/2021/css/custom_property_registered.sql

```diff
@@ -0,0 +1,53 @@
+#standardSQL
+CREATE TEMPORARY FUNCTION getCustomPropertiesWithComputedStyle(payload STRING) RETURNS
+ARRAY<STRING> LANGUAGE js AS '''
+try {
+  var $ = JSON.parse(payload);
+  var vars = JSON.parse($['_css-variables']);
+
+  function walkElements(node, callback) {
+    if (Array.isArray(node)) {
+      for (let n of node) {
+        walkElements(n, callback);
+      }
+    }
+    else {
+      callback(node);
+
+      if (node.children) {
+        walkElements(node.children, callback);
+      }
+    }
+  }
+
+  let ret = new Set();
+
+  walkElements(vars.computed, node => {
+    if (node.declarations) {
+      for (let property in node.declarations) {
+        let o = node.declarations[property];
+
+        if (property.startsWith("--") && o.type) {
+          ret.add(property);
+        }
+      }
+    }
+  });
+
+  return [...ret];
+} catch (e) {
+  return [];
+}
+''';
+
+SELECT DISTINCT
+  _TABLE_SUFFIX AS client,
+  prop,
+  COUNT(DISTINCT url) OVER (PARTITION BY _TABLE_SUFFIX, prop) AS pages,
+  COUNT(DISTINCT url) OVER (PARTITION BY _TABLE_SUFFIX) AS total,
+  COUNT(DISTINCT url) OVER (PARTITION BY _TABLE_SUFFIX, prop) / COUNT(DISTINCT url) OVER (PARTITION BY _TABLE_SUFFIX) AS pct
+FROM
+  `httparchive.pages.2021_07_01_*`,
+  UNNEST(getCustomPropertiesWithComputedStyle(payload)) AS prop
```

Comment:
Potenital UNNEST counting issue

Reply:
WAI: `DISTINCT url`

Reply:
Agree on second look. Happy to leave as resolved.